app/src/main/java/ch/hikemate/app/model/route/HikesViewModel.kt

```diff
@@ -706,7 +767,7 @@ class HikesViewModel(
     }
 
     // Set the hike's planned date to the right one
-    val newSavedHike = SavedHike(id = hike.id, name = hike.name ?: "", date = date)
+    val newSavedHike = hike.toSavedHike()
```

Comment:
Warning: You can't use the `toSavedHike` method here, since the date is not the one from the hike

Reply:
Good catch, it actually showed me that my tests did not notice this, so I added a regression test. Well, not a regression test per se, but I modified the existing ones to check for this mistake.

Reply:
Thanks for the edit!